File: benchmark/corpus/python/sqli_vuln_1.py
"""VULN: CWE-89 — SQL injection via string concatenation (sqlite3)."""
import sqlite3


def find_user(username):
    conn = sqlite3.connect("app.db")
    cursor = conn.cursor()
    query = "SELECT * FROM users WHERE username = '" + username + "'"
    cursor.execute(query)
    return cursor.fetchone()


def login(username, password):
    user = find_user(username)
    if user and user[2] == password:
        return True
    return False

File: benchmark/corpus/java/SqliVuln1.java
// VULN: CWE-89 — SQL injection via Statement.executeQuery() with concatenation
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;

public class SqliVuln1 {

    public static String findUser(String username) throws Exception {
        Connection conn = DriverManager.getConnection(
                "jdbc:mysql://localhost/app", "root", "");
        Statement stmt = conn.createStatement();
        String query = "SELECT email FROM users WHERE username = '" + username + "'";
        ResultSet rs = stmt.executeQuery(query);
        return rs.next() ? rs.getString("email") : null;
    }

    public static void main(String[] args) throws Exception {
        System.out.println(findUser(args[0]));
    }
}
